### PR TITLE
Gruvbox theme

### PR DESCRIPTION
**Pull request by AckslD**
Hi @t-wissmann!

I've been customising my barpyrus panel and thought I would share it here. I'm not sure what's the best way to share customisations and if you don't want it in the repo. Anyway, the bar contains indicators for:
* Mail
* Wifi
* Volume
* Disk space
* CPU
* RAM
* Battery
* Uptime

I've also changed the colorscheme with inspiration from the [gruvbox](https://github.com/morhetz/gruvbox) theme. Not that I also made some changes in `barpyrus/hlwm.py` to do this. It looks as follows, with the matching theme in my terminal and vim:

Screenshot from 2020-04-26 15-54-18

Let me know what you think and if you would want this in the repo or not :)

### CODE REVIEW

File: share/gruvbox_config.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+(x, y, monitor_w, monitor_h) = hc.monitor_rect(monitor)
+height = 14 # height of the panel
+width = monitor_w # width of the panel
+hc(['pad', str(monitor), str(height)]) # get space for the panel
+
+# Conky setup
+custom = ''
+
+mail_symb = '\ue1a8'
+inbox = '/home/axel/.mail/INBOX'
```
Comment (t-wissmann) [inline share/gruvbox_config.py]: I think one should replace this by
```
inbox = os.path.join(os.environ['HOME'], '.mbox/inbox')
```
(also requires `import os` above)

File: barpyrus/colors.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+GREEN_LIGHT = '#b8bb26'
+YELLOW_LIGHT = '#fabd2f'
+BLUE_LIGHT = '#83a598'
+PURPLE_LIGHT = '#d3869b'
+AQUA_LIGHT = '#8ec07c'
+GRAY_LIGHT = '#928374'
+ORANGE_LIGHT = '#fe8019'
+
+def col_fmt(color):
+    return '%{F\\' + color + '}'
```
Comment (t-wissmann) [inline barpyrus/colors.py]: This is not a general barpyrus color formatting but it only works in conky, so `col_fmt` needs to go to `conky.py`.

**Comment by t-wissmann** [comment]
Thanks for sharing your config! I've added a few comments above.

I also think your config should be moved to `share/examples`. I'd like to keep the examples in `share/` serve as entry points for new users and as general as possible (without paths/resource names hard-coded, etc). But it's very valuable to have a collection of more advanced configuration :)

File: share/gruvbox_config.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+net_down_speed = col_fmt(RED_LIGHT) + net_down_speed_symb + col_fmt(FG) + '${downspeedf}K'
+net_up_speed_symb = '\ue13b'
+net_down_speed = col_fmt(GREEN_LIGHT) + net_up_speed_symb + col_fmt(FG) + '${upspeedf}K'
+
+########
+# wifi #
+########
+wifi_icons = ['\ue218', '\ue219', '\ue21a']
+wifi_delta = 100 / len(wifi_icons)
+wifi_qual = '${wireless_link_qual_perc wlp0s20f3}'
```
Comment (t-wissmann) [inline share/gruvbox_config.py]: Can we get rid of this `wlp0s20f3` to make it work for others? In my personal barpyrus config I'm simply listing all network devices: https://github.com/t-wissmann/dotfiles/blob/master/config/barpyrus/config.py#L38..L62

File: share/gruvbox_config.py (hunk 1 above)
Comment (The-Compiler) [reply share/gruvbox_config.py]: I'd vote for `os.path.expanduser('~/.mbox/inbox')` instead :wink: (needs `import os.path`)

Comment (t-wissmann) [reply share/gruvbox_config.py]: Yes, that's much better!

Comment (AckslD) [reply share/gruvbox_config.py]: Woops, indeed!

**Comment by AckslD** [comment]
@t-wissmann, thanks for your comments, I'll update the PR later today

File: barpyrus/colors.py (hunk 2 above)
Comment (AckslD) [reply barpyrus/colors.py]: Ah okay!

File: share/gruvbox_config.py (hunk 3 above)
Comment (AckslD) [reply share/gruvbox_config.py]: I'll check how you do it

File: share/gruvbox_config.py (hunk 1 above)
Comment (AckslD) [reply share/gruvbox_config.py]: Done

File: share/gruvbox_config.py (hunk 3 above)
Comment (AckslD) [reply share/gruvbox_config.py]: I now use the first entry in /sys/class/net which starts with 'w' and if none exists I show a disconnected symbol

File: barpyrus/colors.py (hunk 2 above)
Comment (AckslD) [reply barpyrus/colors.py]: Done

**Comment by AckslD** [comment]
@t-wissmann I fixed the things you pointed out. Btw something that should maybe still be done and it depends a bit on how you want to handle it. At the moment I have edited `barpyrus/hlwm.py` to edit colors of some things. However it is still not configurable from config.py. What's your view on this?

**Comment by t-wissmann** [comment]
Sorry for not writing earlier... I completely forgot about the open PRs in for barpyrus.. The color-settings look good, and I think one can configure the colors by adjusting the global variables. I think this is fine for the moment!

I'm adding some more comments. If it's fine for you I can also fix them, given that I was letting you wait for so long.

File: barpyrus/lemonbar.py [hunk 4]
```diff
@@ -28,9 +29,12 @@ def __init__(self, geometry = None,
         command += '-a 100 -d -u 2'.split(' ')
         command += [ '-B', background  ]
         command += [ '-F', foreground  ]
+        command += [ '-o 0' ]
         command += [ '-f', font  ]
+        command += [ '-o 0' ]
         command += '-f -*-*-*-*-*-*-2-*-*-*-*-*-*-*'.split(' ')
         if symbol_font != None:
+            command += [ '-o -%d' % symbol_vert_offset ]
```
Comment (t-wissmann) [inline barpyrus/lemonbar.py]: The lemonbar I'm using here does not know the `-o` flag you were passing to it. Looking at  lemonbar's source on https://github.com/LemonBoy/bar/blob/master/lemonbar.c#L1461 it seems that its about the randr output, so I think it does not belong to the general code. If you need it, we can add an `additional_args=[]` as a parameter to `__init__` such that you can pass it around from your config.py

File: barpyrus/widgets.py [hunk 5]
```diff
@@ -44,7 +44,11 @@ def next_timeout(self):
     def maybe_timeout(self, now):
         some_timeout = False
         for w in self.subwidgets:
-            some_timeout = w.maybe_timeout(now) or some_timeout
+            try:
+                some_timeout = w.maybe_timeout(now) or some_timeout
+            except:
+                breakpoint()
+                raise RuntimeError()
```
Comment (t-wissmann) [inline barpyrus/widgets.py]: I think this was just for you and does not belong to this PR, doesn't it?

Comment (AckslD) [reply barpyrus/widgets.py]: Woops, indeed

File: barpyrus/lemonbar.py (hunk 4 above)
Comment (AckslD) [reply barpyrus/lemonbar.py]: Hmm, I see, so I was using `lemonbar-xft` with this option: https://github.com/drscream/lemonbar-xft/blob/xft-port/lemonbar.c#L1494

**Comment by AckslD** [comment]
Hi @t-wissmann! No problem :)

File: barpyrus/lemonbar.py (hunk 4 above)
Comment (t-wissmann) [reply barpyrus/lemonbar.py]: I see, thanks! Since adding this is part of #11 I'd remove it from the present PR.

**Comment by t-wissmann** [comment]
@AckslD I took the freedom and edited the changes discussed above directly to this PR. (I hope you were not already working on it....) If it's fine for you, I can merge this PR as is

**Comment by AckslD** [comment]
Cool @t-wissmann, go ahead and merge :)

**Comment by AckslD** [comment]
Btw, just leaving a reference here for an updated version of the config that I've added things for from time to time and can be found [here](https://gitlab.com/AckslD/config/-/blob/master/barpyrus/config.py). It also has support for showing the current song playing (using playerctl and e.g. ncspot), clicking the wifi icon to see the current wifi name, toggling a onscreen-keyboard etc :)

**Comment by t-wissmann** [comment]
OK, thanks! :)
